File: PokeLangPython/backup/py_AFD/ID.py
from _AFD import AFD
from Token import Token

class Functions:
    validacoes = [' ', '+', '-', '*', '/', '(', ')', '{', '}', ';', '\n', None]

    @staticmethod
    def validate_end(code):
        return code[0] in Functions.validacoes

class ID(AFD):
    def evaluate(self, code):
        if code[0].isalpha():
            letter = self.read_letter(code)

            if Functions.validate_end(code):
                return Token("ID", letter)

        return None

    def read_letter(self, code):
        letter = ""
        while code[0].isalpha():
            letter += code[0]
            code.next()
        return letter

File: src/AFD/ID.java
package AFD;

import java.text.CharacterIterator;
import java.util.Arrays;
import java.util.List;
import Token.Token;

public class ID extends AFD {
    @Override
    public Token evaluate(CharacterIterator code) {
        if (Character.isLetter(code.current())) {
            String letter = readID(code);

            if (endID(code) == true) {
                return new Token("id", letter);
            }
        }

        return null;
    }

    private String readID(CharacterIterator code) { // Lê a entrada e verifica se é uma letra, se for, incrementa em "letter"
        String letter = "";

        while (Character.isLetter(code.current())) {
            letter += code.current();
            code.next();
        }
        return letter;
    }

    private boolean endID(CharacterIterator code) { // Verifica se finalizou a declaração do ID
        List<Character> finalizadores = Arrays.asList(' ', '+', '-', '*', '/', '(', ')', '{', '}', '[', ']', ';', '\n' , CharacterIterator.DONE);

        for (char finalizador : finalizadores) {
            if (code.current() == finalizador) {
                return true;
            }
        }
        return false;
    }
}
